### DISCUSSION TITLE
Firefly Importer not working

### DISCUSSION THREAD

**Discussion by veniplex** [firefly-iii/firefly-iii#7252] [Firefly III]
Hi,

I tried now several times to set up the FIDI, but nevertheless what I do it doesn't work.
I followed the whole documentation step-by-step. I even reinstalled everything several times now.

I am using Docker on a Synology NAS and the newest versions of Firefly / FIDI.

I am able to set up FIDI and get to the page where I can authenticate, but thats it. After that its not working.

So the problem I always run back into, is:
1. If I try to authenticate via Client ID (regardless if entered in the ENV's or over the web interface). After the authemtication seems to have worked, because I can see an entry in the Firefly settings, I get redirected to the FIDI but the url looks like this `https://firefly-importer.mydomain.com/callback?code=xxx`, where `xxx` is a long string.
2. If I try to use the method via personal access token, I am always automatically redirected to `https://firefly-importer.mydomain.com/token` and I get a page from Synology NAS which says "Could not find requested site".

I don't understand what the problem is or what I am doing wrong. Has anyone an idea?

Thanks.

**Comment by JC5** [comment]
> were xxx is a long string.

So on this page you get an error or something?

**Reply by veniplex** [reply]
Yes. When the callback happens I see an error page from Synology saying "Page could not be found.". If I go then back to the base url of FIDI, it asks again for authentication per Firefly III URL and Client ID.

**Reply by JC5** [reply]
A possibility is that the URL is too long. What happens when you start cutting of info from the URL until it gives a different error?

**Reply by veniplex** [reply]
Okay, so I tried it now. Here are my steps:

1. I go to the FIDI page (e.g. `https://firefly-importer.mydomain.com`).
2. It redirects me automatically to `https://firefly-importer.mydomain.com/token`. Just to be sure, I press **Start over** and **Reauthenticate**.
3. It asks for Firefly III URL and Client ID.
4. For Firefly III URL I enter `https://firefly.mydomain.com`.
5. For Client ID I go to `https://firefly.mydomain.com/profile` and then to the OAuth-Tab. I generate a new client as described in the docs. I make sure that there is only a Client ID visible and no authorised Apps or generated personal tokens.
6. I go back to `https://firefly-importer.mydomain.com/token` and enter the Client ID in the field below the Firefly III (e.g. `4`).
7. I click on `Submit`.
8. I get forwarded to `https://firefly.mydomain.com/oauth/authorize?client_id=4&redirect_uri=https%3A%2F%2Ffirefly-importer.mydomain.com%2Fcallback&response_type=code&scope=&state=rwiNFiLo9zuyzOJ4bcnx2EvNZ4sEh6foOfsKUSV0&code_challenge=LzB9_ppxitBD4xAKeg68TPv9d6_L3m53H5aN8M9OmrU&code_challenge_method=S256`.
9. I click on allow.
10. I get redirected to `https://firefly-importer.mydomain.com/callback?code=def50200aebd24a61caa8b7cf0703b4b66c15eed0d701ee17e6b3cbeb8f5c3edd9a84130f69d657fa79b92fbe2d703ff9ca3aef1da0f9c5003798284833ec4bb645f980643ea59e8c912af31a9256fe4994beaa020083c10f0e0ffd5deb2f67409141965642ee6b69d21ed8781ab9213205086e706f0caa257b4104b22278d38479776dacffb9a5df91072db82ce6b6d9d28c6fddc4ab967569b7053dfcb207803eac9a18581dbc9b8ffba9dff6c8a6aa4d06f142e1a441ed09acb381d6d647101ca37f6212e6801f64a7577b77dc6556c95bcad17185af0e72c4a08a1742f7d5314d5130e3c055961003b2b1b310c5a3c909dab750bc68ff150c90a79604168a990544b989564b78b4323457048d997253b27a9e15c1cc5dff080dad779e12acbaecbd26b814ba29a792f4545c08c82d7de21f1b73e7960fa82379ca284584f8489adf6a578c112f165f74c837d0b879848651d07d75e22209885870b564a6fbd3c98d5064c94e58ea49d6165218d651f2e4f7b60d3ef1f01a0d62b497037816e86d6c8bdb50bf8305666e4bd47b6567db04c4f69&state=rwiNFiLo9zuyzOJ4bcnx2EvNZ4sEh6foOfsKUSV0`.
11. I see the error page from Synology, that this page cannot be found. [Screenshot](https://imgur.com/a/q6MUJo9)
12. I check `https://firefly.mydomain.com/profile` and see that there is a new authorised app `Firefly Importer`.
13. I go back to the browser tab with the url from step 10.
14. I remove characters at the end of the url and I get a new error message from FIDI. [Screenshot](https://imgur.com/a/ggjhEY6)
```
Whoops! 500 :(
Sorry, the Firefly III Data Importer broke down.

Error message
The "state" returned from your server doesn't match the state that was sent.

More information
The error occurred in /var/www/html/app/Http/Controllers/TokenController.php:74.

Please collect more information in the storage/logs directory, where you will find log files. If you're running Docker, use docker logs -f [container]. You can read more about collecting error information [in the FAQ](https://docs.firefly-iii.org/faq/other#how-do-i-enable-debug-mode).

Get help on GitHub
You're more than welcome to open a new issue [on GitHub](https://github.com/firefly-iii/firefly-iii/issues).

Use the search!
Include the information [from this debug page](https://firefly-importer.mydomain.com/debug).
Tell us more than "it says Whoops!"
Include error logs (see above).
Tell us what you were doing.
```
15. I go to the debug page on `https://firefly-importer.mydomain.com/debug` and see this:
Debug information generated at 2023-03-20 11:00:14 Europe/Berlin for Firefly III Data Importer version **1.2.2**.

| Scope | Version |
| --- | --- |
| FIDI | 1.2.2 |
| Build | 205, 18-03-2023 14:56:24 CET |
| PHP | 8.2.3 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| System TZ | Europe/Berlin |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36 |
16. I have no ad-blocker or something like that enabled.

**Reply by JC5** [reply]
All I can imagine is that the webserver of Synology (ie. the reverse proxy) can't handle URLs that are too long. Unfortunately, I have no solution :(

**Reply by veniplex** [reply]
So far I didn't found any url limits. I will try to dig deeper. If i find a solution I will post a comment :)... Thank you so far for your help.
